app/models/product_import/products_reset_strategy.rb

```diff
@@ -0,0 +1,34 @@
+module ProductImport
+  class ProductsResetStrategy
+    def initialize(excluded_items_ids)
+      @excluded_items_ids = excluded_items_ids
+    end
+
+    def reset(supplier_ids)
+      @supplier_ids = supplier_ids
+
+      if supplier_ids.present?
+        relation.update_all(count_on_hand: 0)
+      else
+        0
```

Comment:
we wouldn't need this conditional branch if we were in Rails 4.2 :sob: See https://stackoverflow.com/questions/4877931/how-to-return-an-empty-activerecord-relation#10001043